### PR TITLE
CHE-1950: Add foreign reference on account to Workspace

### PR DESCRIPTION
**Pull request by akorneta**
### What does this PR do?

Adds a constraint on workspace and defines relation between workspace and account.
### What issues does this PR fix or reference?
#1950
### New Behavior

During create/update of the workspace with non existing namespace(account) exception will be thrown.
During remove an account, all the workspaces related to this account will be removed.

@sleshchenko, @evoevodin, @skabashnyuk please take a look.

### CODE REVIEW

File: wsmaster/che-core-api-workspace/src/test/java/org/eclipse/che/api/workspace/server/DefaultWorkspaceValidatorTest.java [hunk 1]
```diff
@@ -126,33 +126,33 @@ public void shouldValidateCorrectWorkspaceName(String name) throws Exception {
...
-        workspace.getAttributes().put("codenvy_key", "value1");
-
-        wsValidator.validateWorkspace(workspace);
-    }
+//    @Test(expectedExceptions = BadRequestException.class,
+//          expectedExceptionsMessageRegExp = "Attribute name 'null' is not valid")
+//    public void shouldFailValidationIfAttributeNameIsNull() throws Exception {
+//        final WorkspaceImpl workspace = new WorkspaceImpl("id", "namespace", createConfig());
+//        workspace.getAttributes().put(null, "value1");
+//
```
Comment (skabashnyuk) [inline wsmaster/che-core-api-workspace/src/test/java/org/eclipse/che/api/workspace/server/DefaultWorkspaceValidatorTest.java]: why this tests are commented?

**Comment by skabashnyuk** [comment]
ok @evoevodin ?

**Comment by sleshchenko** [comment]
LGTM

**Comment by voievodin** [comment]
Ok

**Pull request merged**